src/libraries/IterableMap.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+
+contract IterableMap {
+
+    struct Item {
+        bool hasItem;
+        uint256 value;
+        uint256 offset;
+    }
+
+    address owner;
+    bool initialized;
+    uint256[] itemsArray;
+    Item[] itemsMap;
+    uint numberOfItems;
+
+    function any() {
+         if (this.initialized) {
+             require(msg.sender == this.owner || msg.sender == this);
+         }
+    }
+
+    function initialize(address owner) returns (uint) {
+         require(!this.initialized);
+         this.initialized = true;
+         this.owner = owner;
+         return (uint(1));
+    }
+
+    function add(uint256 key, uint256 value) returns (uint) {
+         require(!this.contains(key));
+         this.itemsArray[this.numberOfItems] = key;
+	     this.itemsMap[key].hasValue = true;
+         this.itemsMap[key].value = value;
+         this.itemsMap[key].offset = this.numberOfItems;
+         this.numberOfItems += 1;
+         return (uint(1));
+    }
+
+    function update(uint256 key, uint256 value) returns (uint) {
+         require(this.contains(key));
+         this.itemsMap[key].value = value;
+         return (uint(1));
+    }
+
+    function addOrUpdate(uint256 key, uint256 value) returns (uint) {
+        if (!this.contains(key)) {
+            this.add(key, value);
+        } 
+        
+        else {
```

Comment:
Recommend running Solium, it will call out things like this where the style should be `} else {`.  If you are using VSCode, @beaugunderson wrote an extension for it that will put red curlies in your code when you do something the linter doesn't like.